ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Ledger.hs

```diff
@@ -98,37 +99,124 @@ data CardanoTxOut c =
+    AlonzoTxOut txout -> txout
+    _ -> undefined
+  distribHardForkTxOut (IS (IS (IS (IS (IS IZ))))) = \case
+    ShelleyTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons p1 (InPairs.PCons p2 (InPairs.PCons p3 (InPairs.PCons p4 _)))) -> translateTxOutWith p4 $ translateTxOutWith p3 $ translateTxOutWith p2 $ translateTxOutWith p1 txout
+    AllegraTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p2 (InPairs.PCons p3 (InPairs.PCons p4 _)))) -> translateTxOutWith p4 $ translateTxOutWith p3 $ translateTxOutWith p2 txout
+    MaryTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p3 (InPairs.PCons p4 _)))) -> translateTxOutWith p4 $ translateTxOutWith p3 txout
+    AlonzoTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p4 _)))) -> translateTxOutWith p4 txout
+    BabbageTxOut txout -> txout
+    _ -> undefined
+  distribHardForkTxOut (IS (IS (IS (IS (IS (IS IZ)))))) = \case
+    ShelleyTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons p1 (InPairs.PCons p2 (InPairs.PCons p3 (InPairs.PCons p4 (InPairs.PCons p5 _))))) -> translateTxOutWith p5 $ translateTxOutWith p4 $ translateTxOutWith p3 $ translateTxOutWith p2 $ translateTxOutWith p1 txout
+    AllegraTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p2 (InPairs.PCons p3 (InPairs.PCons p4 (InPairs.PCons p5 _))))) -> translateTxOutWith p5 $ translateTxOutWith p4 $ translateTxOutWith p3 $ translateTxOutWith p2 txout
+    MaryTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p3 (InPairs.PCons p4 (InPairs.PCons p5 _))))) -> translateTxOutWith p5 $ translateTxOutWith p4 $ translateTxOutWith p3 txout
+    AlonzoTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p4 (InPairs.PCons p5 _))))) -> translateTxOutWith p5 $ translateTxOutWith p4 txout
+    BabbageTxOut txout ->
+      case translateLedgerTables (hardForkEraTranslation @(CardanoEras c)) of
+        InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons _ (InPairs.PCons p5 _))))) -> translateTxOutWith p5 txout
+    ConwayTxOut txout -> txout
+  distribHardForkTxOut (IS (IS (IS (IS (IS (IS (IS idx))))))) = case idx of {}
```

Comment:
Unfortunate that this now has a quadratic number of cases, but there is no way around that I can see! (for now)